cmd/vela-terraform/init.go

```diff
@@ -0,0 +1,221 @@
+		// add flag for ForceCopy from provided init command
+		flags = append(flags, "-force-copy")
+	}
+
+	// check if FromModule is provided
+	if len(i.InitOptions.FromModule) > 0 {
+		// add flag for FromModule from provided init command
+		flags = append(flags, fmt.Sprintf("-from-module=%s", i.InitOptions.FromModule))
+	}
+
+	// check if Get is provided
+	if i.InitOptions.Get {
+		// add flag for Get from provided init command
+		flags = append(flags, "-get=true")
+	}
+
+	// check if GetPlugins is provided
+	if i.InitOptions.GetPlugins {
+		// add flag for GetPlugins from provided init command
+		flags = append(flags, "-get-plugins=true")
+	}
+
+	// check if Input is provided
+	if i.InitOptions.Input {
+		// add flag for Input from provided init command
+		flags = append(flags, "-input=true")
+	}
+
+	// check if Lock is provided
+	if i.InitOptions.Lock {
+		// add flag for Lock from provided init command
+		flags = append(flags, "-lock=true")
+	}
+
+	// check if LockTimeout is provided
+	if i.InitOptions.LockTimeout > 0 {
+		// add flag for LockTimeout from provided init command
+		flags = append(flags, fmt.Sprintf("-lock-timeout=%s", i.InitOptions.LockTimeout))
+	}
+
+	// check if NoColor is provided
+	if i.InitOptions.NoColor {
+		// add flag for NoColor from provided init command
+		flags = append(flags, "-no-color")
+	}
+
+	// check if PluginDirs is provided
+	if len(i.InitOptions.PluginDirs) > 0 {
+		var configs string
+		for _, v := range i.InitOptions.PluginDirs {
+			configs += fmt.Sprintf("-plugin-dir=%s ", v)
+		}
+
+		// add flag for PluginDirs from provided init command
+		flags = append(flags, strings.TrimSuffix(configs, " "))
+	}
+
+	// check if Reconfigure is provided
+	if i.InitOptions.Reconfigure {
+		// add flag for Reconfigure from provided init command
+		flags = append(flags, "-reconfigure")
+	}
+
+	// check if Upgrade is provided
+	if i.InitOptions.Upgrade {
+		// add flag for Upgrade from provided init command
+		flags = append(flags, "-upgrade=false")
+	}
+
+	// check if VerifyPlugins is provided
+	if i.InitOptions.VerifyPlugins {
+		// add flag for VerifyPlugins from provided init command
+		flags = append(flags, "-verify-plugins=true")
+	}
+
+	// add the required dir param
+	flags = append(flags, dir)
+
+	return exec.Command(_terraform, append([]string{initAction}, flags...)...)
+}
+
+// Exec formats and runs the commands for initing Terraform.
+func (i *Init) Exec() error {
+	logrus.Trace("running init with provided configuration")
+
+	// create the init command for the file
+	cmd := i.Command(i.Directory)
+
+	// run the init command for the file
+	err := execCmd(cmd)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Validate verifies the Init is properly configured.
+func (i *Init) Validate() error {
+	logrus.Trace("validating init plugin configuration")
+
+	if len(i.Directory) == 0 {
+		logrus.Warn("terrafrom init will run in current dir")
+
+		// set the directory to run in current dir
+		i.Directory = "."
```

Comment:
would it make sense to set the default in `main.go` where we define all the flags?

Reply:
I think that would be fine. Would you still want to keep the warning message or assume since it's a default they understand the behavior? The main reason I did it there was the warning message. Which I can still do if we default, just curious if you see value in it.

Reply:
different audiences i think, so i would keep both. imo, it's more discoverable to put the default where the flag is defined for developers or the curious. and returning the warning is more-so to provide transparency for the end-user.

Reply:
Awesome! I agree I like having the warning for people. it's all updated